layers/+lang/ocaml/packages.el

```diff
@@ -32,18 +32,16 @@
       :modes merlin-mode
       :variables merlin-completion-with-doc t)))
 
-(when (configuration-layer/layer-used-p 'syntax-checking)
-  (defun ocaml/post-init-flycheck ()
-    (spacemacs/enable-flycheck 'tuareg-mode))
-  (defun ocaml/init-flycheck-ocaml ()
-    (use-package flycheck-ocaml
-      :if (configuration-layer/package-used-p 'flycheck)
-      :defer t
-      :init
-      (progn
-        (with-eval-after-load 'merlin
-          (setq merlin-error-after-save nil)
-          (flycheck-ocaml-setup))))))
+(defun ocaml/post-init-flycheck ()
+  (spacemacs/enable-flycheck 'tuareg-mode))
+
+(defun ocaml/init-flycheck-ocaml ()
+  (defun ocaml/setup-flycheck ()
+    (when (and buffer-file-name (locate-dominating-file buffer-file-name ".merlin"))
+      (setq merlin-error-after-save nil)
+      (flycheck-ocaml-setup)))
+  (use-package flycheck-ocaml
+    :hook (merlin-mode . ocaml/setup-flycheck)))
```

Comment:
What is `ocaml/setup-flycheck`? I can't find its definition.

Reply:
It is part of this PR: https://github.com/syl20bnr/spacemacs/blob/4d14a80847202a3c003848d728fdb9ed831238d7/layers/%2Blang/ocaml/packages.el#L39-L42
It disables merlin's own on-save error reporting (since flycheck-ocaml will report it now), and then activates flycheck if a `.merlin` is present.

Reply:
Lol. Silly me. Thanks for pointing out.